themes/V3/5ePHB/snippets.js

```diff
@@ -29,6 +30,49 @@ module.exports = [
 				icon : 'fas fa-sort-numeric-down',
 				gen  : '{{pageNumber,auto}}\n{{footnote PART 1 | SECTION NAME}}\n\n'
 			},
+			{
+				name        : 'Footer',
+				icon        : 'fas fa-shoe-prints',
+				gen         : FooterGen.createFooter,
+				subsnippets : [
+					{
+						name   : 'Footer from H1',
+						icon   : 'fas fa-dice-one',
+						gen    : FooterGen.createFooter,
+						params : { headerSize: 1 }
```

Comment:
Could I suggest this format? Then no need to pass around a params object all the way to the editor and back.

```suggestion
						gen    : FooterGen.createFooter(1),
```

Reply:
I didn't think I could actually use `FooterGen.createFooter(1)` due to the parameters, but I think I just came up with a solution. I will experiment and update the PR if my madness turns out to be genius.

Reply:
I'll push a new commit in a moment, but I've changed the generation function to return a function that takes a props parameter. By passing a parameter (headerSize) to the generation function, it returns a footer function that searches for first instance of that many `#` followed by a ` ` at the start of a preceding line.
It took a bit to wrap my brain around how to pass the `headerSize` parameter from the generation function into the returned function, but it looks to be working in my dev branch now.

Reply:
I've also shifted to passing the entire `this.props` rather than `this.props.brew` and `this.props.cursorPos` as separate parameters. Hopefully having access to the entirety of the props will be useful future snippets.